### PR TITLE
Rewrite of Quick Start

### PR DESCRIPTION
**Pull request by NWeber-SL**
Review specifically:

* Updated info on HelloShiftLeft to include that it is in Ocular distribution

### CODE REVIEW

File: using-ocular/ocular-desc.md [hunk 1]
```diff
@@ -1,40 +1,19 @@
-# ShiftLeft Ocular
+# About ShiftLeft Ocular
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-desc.md]: We should rename the file name to match this title -> `about-shiftleft-ocular.md`.

File: using-ocular/ocular-desc.md [hunk 2]
```diff
@@ -1,40 +1,19 @@
...
+* **Generation of [Code Property Graphs (CPG)](../introduction/understanding-cpg.md).** (Included in the ShiftLeft Ocular trial version). Based on your application code, in the form of either a JAR for Java or as source code for C/C++, ShiftLeft Ocular generates a CPG that can be used as the basis of custom static analysis. CPGs provide an extensible, multi-layered representation of each unique code version. CPGs are serializable to various output/exchange formats.
+
+* **Interactive Code Analysis Using a Read-Eval-Print-Loop (REPL).** (Included in the ShiftLeft Ocular trial version). CPGs and Security Profiles can be interactively explored using a REPL, an interactive shell with support for the Ocular Query Language (OQL). Among other features, the REPL offers utilities for exporting security analysis results as plain text or in JSON format, readline support, and tab-completion.
 
+* **Custom Analysis Scripts and Extensibility.** Included in the ShiftLeft Ocular trial version). User-provided REPL scripts can be executed non-interactively to perform automated custom scans for security issues. Moreover, you can even augment and customize OQL via the "pimp-my-library" pattern.
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-desc.md]: I believe we wanted to reword "pimp-my-library"?

File: using-ocular/ocular-desc.md [hunk 3]
```diff
@@ -1,40 +1,19 @@
...
 
-* [ShiftLeft Ocular Demo by Fabian Yamaguchi](https://www.youtube.com/watch?v=n_dpgI2RhEU)
-* [ShiftLeft Ocular Demo by Chetan Conikee](https://www.youtube.com/watch?v=bB2C-FzC1Yw)
+* **Automatic Code Scanning through Security Profiles.** Policies can be applied to CPGs to generate Security Profiles, which are summaries of vulnerabilities and data leaks present in the code. The Security Profile is part of the CPG, as an [overlay](https://ocular.shiftleft.io/api/io/shiftleft/repl/cpgcreation/Overlays$.html). Security Profiles can be explored and processed using the REPL and managed in a workspace.
+  
+* **[Workspaces](workspaces.md) for Managing CPGs and Security Profiles.** A workspace is a directory containing all data persisted by ShiftLeft Ocular. Workspaces allow you to easily manage CPGs and Security Profiles, load and simultaneously work with more than one CPG in a given workspace, and combine queries.
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/ocular-desc.md]: To the user, the `workspace` should not be a directory. The directory is an implementation detail.

To the user, the workspace should be a container for managing CPGs and SPs.

File: using-ocular/getting-started/ocular-quick-start.md [hunk 4]
```diff
@@ -1,71 +1,26 @@
...
-
-**Note** This option affects all subsequent JVM instances started on the machine.
-
-In order to restrict the JVM option to only the parts of ShiftLeft Ocular that you intend to use, you can also provide the memory allocation individually. For example, to set 12 GB available heap memory to `java2cpg` or `ocular`, use this option directly:
-
-```bash
-./ocular.sh -J-Xmx12g
-./java2cpg.sh -J-Xmx12g <...>
-```
+The demo Java application [HelloShiftLeft](https://github.com/ShiftLeftSecurity/HelloShiftLeft) is used in this Quick Start. It is included in the ShiftLeft Ocular distribution. When you install ShiftLeft Ocular, HelloShiftLeft is automatically placed in the `subjects` subdirectory. The file is `hello-shiftleft-0.0.1-SNAPSHOT.jar`.
```
[the remark this answers is not in the document]
Comment (NWeber-SL) [reply using-ocular/getting-started/ocular-quick-start.md]: Done

File: using-ocular/getting-started/ocular-quick-start.md [hunk 5]
```diff
@@ -105,40 +60,54 @@ try {
...
 In this code fragment, a cookie is received via HTTP and eventually deserialized to create a Java object, an optimistic practice that is often exploited by attackers for arbitrary code execution.
 
-## Generating the CPG
+## Installing ShiftLeft Ocular
+
+Refer to the article [Installing ShiftLeft Ocular](installation.md) for details.
+
+## Starting ShiftLeft Ocular
+
+Information for the supported operating systems can be found in the article [Starting ShiftLeft Ocular](starting.md)
```
[the remark this answers is not in the document]
Comment (NWeber-SL) [reply using-ocular/getting-started/ocular-quick-start.md]: Fixed

File: using-ocular/getting-started/ocular-quick-start.md [hunk 6]
```diff
@@ -105,40 +60,54 @@ try {
...
-cd $shiftleft
-./ocular.sh
+```
 createCpg("subjects/hello-shiftleft-0.0.1-SNAPSHOT.jar")
 ```
 
-This command creates a CPG for the HelloShiftLeft application and stores it in the workspace. Issuing the command `workspace` shows a workspace entry for the newly created CPG.
+where `subjects/hello-shiftleft-0.0.1-SNAPSHOT.jar` is the location and filename of the target application.
+
+The file cpg.bin.zip in a binary format, containing the CPG for the target application, is created and automatically loaded into your workspace.
```
[the remark this answers is not in the document]
Comment (NWeber-SL) [reply using-ocular/getting-started/ocular-quick-start.md]: Done

File: using-ocular/getting-started/ocular-quick-start.md [hunk 7]
```diff
@@ -105,40 +60,54 @@ try {
...
 ```
 
-This command creates a CPG for the HelloShiftLeft application and stores it in the workspace. Issuing the command `workspace` shows a workspace entry for the newly created CPG.
+where `subjects/hello-shiftleft-0.0.1-SNAPSHOT.jar` is the location and filename of the target application.
+
+The file cpg.bin.zip in a binary format, containing the CPG for the target application, is created and automatically loaded into your workspace. 
+
+## Exploring Your Workspace
+
+The first time you create a CPG, ShiftLeft Ocular automatically creates your workspace. A workspace is a directory containing all data persisted by ShiftLeft Ocular. Workspaces help you manage CPGs and their layers, simultaneously work with multiple CPGs, and combine queries from multiple CPGs.
```
[the remark this answers is not in the document]
Comment (NWeber-SL) [reply using-ocular/getting-started/ocular-quick-start.md]: Done

File: using-ocular/getting-started/ocular-quick-start.md [hunk 8]
```diff
@@ -105,40 +60,54 @@ try {
...
-## Uncovering Attack Surface
+## Uncovering the Attack Surface
 
 The CPG contains information about the processed code on different levels of abstraction: dependencies, type
-hierarchies, control flow, data flow, and instruction-level information. The CPG can be queried interactively using 
-ShiftLeft Ocular or via non-interactive scripts. Using interactive querying, explore the program dependencies by:
+hierarchies, control flow, data flow, and instruction-level information. The CPG can be queried using 
+ShiftLeft Ocular in either [interactive or non-interactive modes](interactive-noninteractive-modes.md). 
+
+Using interactive querying, explore the program dependencies by
```
[the remark this answers is not in the document]
Comment (NWeber-SL) [reply using-ocular/getting-started/ocular-quick-start.md]: Done here. Will create a PR to add to all other articles

File: using-ocular/ocular-desc.md (hunk 2 above)
Comment (NWeber-SL) [reply using-ocular/ocular-desc.md]: Good catch

File: using-ocular/ocular-desc.md (hunk 3 above)
Comment (NWeber-SL) [reply using-ocular/ocular-desc.md]: Done. Fabs particularly wanted "containing all data persisted"

File: using-ocular/about-shiftleft-ocular.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+* **Generation of [Code Property Graphs (CPG)](../introduction/understanding-cpg.md).** (Included in the ShiftLeft Ocular trial version). Based on your application code, in the form of either a JAR for Java or as source code for C/C++, ShiftLeft Ocular generates a CPG that can be used as the basis of custom static analysis. CPGs provide an extensible, multi-layered representation of each unique code version. CPGs are serializable to various output/exchange formats.
+
+* **Interactive Code Analysis Using a Read-Eval-Print-Loop (REPL).** (Included in the ShiftLeft Ocular trial version). CPGs and Security Profiles can be interactively explored using a REPL, an interactive shell with support for the Ocular Query Language (OQL). Among other features, the REPL offers utilities for exporting security analysis results as plain text or in JSON format, readline support, and tab-completion.
+
+* **Custom Analysis Scripts and Extensibility.** Included in the ShiftLeft Ocular trial version). User-provided REPL scripts can be executed non-interactively to perform automated custom scans for security issues. Moreover, you can even augment and customize OQL via the "enhance-my-library" pattern.
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/about-shiftleft-ocular.md]:
```suggestion
* **Custom Analysis Scripts and Extensibility.** (Included in the ShiftLeft Ocular trial version). User-provided REPL scripts can be executed non-interactively to perform automated custom scans for security issues. Moreover, you can even augment and customize OQL via the "enhance-my-library" pattern.
```

File: using-ocular/about-shiftleft-ocular.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+* **Framework and Library Support through Policies.** ShiftLeft Ocular includes Policies, which are annotations for common Java frameworks and libraries. Possibly attacker-controlled data sources and interesting sinks are automatically tagged in the CPG, and flow descriptions exist to scan for common vulnerability patterns. You can provide additional annotations to extend supported frameworks and libraries or encode additional vulnerability patterns.
+
+* **Automatic Code Scanning through Security Profiles.** Policies can be applied to CPGs to generate Security Profiles, which are summaries of vulnerabilities and data leaks present in the code. The Security Profile is part of the CPG, as an [overlay](https://ocular.shiftleft.io/api/io/shiftleft/repl/cpgcreation/Overlays$.html). Security Profiles can be explored and processed using the REPL and managed in a workspace.
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/about-shiftleft-ocular.md]: It's slightly confusing that we say that from CPG + Policy you get the SP. But then we say that the SP is part of the CPG.

Perhaps we can reword a bit, to clarify. We should say that the resulting SP is then incorporated as part of the CPG as an additional layer over the base CPG.

File: using-ocular/getting-started/ocular-quick-start.md [hunk 11]
```diff
@@ -105,43 +62,57 @@ try {
...
 
 ```scala
-cpg.dependency.name.l
+ocular> cpg.dependency.name.l
 ```
 
-A complete list of all dependency names is returned. ShiftLeft Ocular supports functional combinators. For example, to output (name, version) pairs, use the following expression:
+A complete list of all dependency names is returned. ShiftLeft Ocular supports functional combinators. For example, to output (name, version) pairs, use the following expression
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/getting-started/ocular-quick-start.md]:
```suggestion
A complete list of all dependency names is returned. As OQL is a Scala-based DSL, it also supports functional combinators. For example, to output `(name, version)` pairs, use the following expression
```

File: using-ocular/ocular-desc.md (hunk 1 above)
Comment (NWeber-SL) [reply using-ocular/ocular-desc.md]: OK.

File: using-ocular/about-shiftleft-ocular.md (hunk 9 above)
Comment (NWeber-SL) [reply using-ocular/about-shiftleft-ocular.md]: Done

File: using-ocular/about-shiftleft-ocular.md (hunk 10 above)
Comment (NWeber-SL) [reply using-ocular/about-shiftleft-ocular.md]: Done

File: using-ocular/getting-started/ocular-quick-start.md (hunk 11 above)
Comment (NWeber-SL) [reply using-ocular/getting-started/ocular-quick-start.md]: Done

File: using-ocular/about-shiftleft-ocular.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+
+* **Custom Analysis Scripts and Extensibility.** (Included in the ShiftLeft Ocular trial version). User-provided REPL scripts can be executed non-interactively to perform automated custom scans for security issues. Moreover, you can even augment and customize OQL via the "enhance-my-library" pattern.
+
+* **Framework and Library Support through Policies.** ShiftLeft Ocular includes Policies, which are annotations for common Java frameworks and libraries. Possibly attacker-controlled data sources and interesting sinks are automatically tagged in the CPG, and flow descriptions exist to scan for common vulnerability patterns. You can provide additional annotations to extend supported frameworks and libraries or encode additional vulnerability patterns.
+
+* **Automatic Code Scanning through Security Profiles.** Security Profiles are summaries of vulnerabilities and data leaks present in the code. The Security Profile is part of the CPG, as an [layer](https://ocular.shiftleft.io/api/io/shiftleft/repl/cpgcreation/Overlays$.html). Security Profiles can be explored and processed using the REPL and managed in a workspace.
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/about-shiftleft-ocular.md]:
```suggestion
* **Automatic Code Scanning through Security Profiles.** Security Profiles are summaries of vulnerabilities and data leaks present in the code. The Security Profile is part of the CPG, as a [layer](https://ocular.shiftleft.io/api/io/shiftleft/repl/cpgcreation/Overlays$.html). Security Profiles can be explored and processed using the REPL and managed in a workspace.
```

Comment (NWeber-SL) [reply using-ocular/about-shiftleft-ocular.md]: Fixed

**Pull request merged**